### PR TITLE
feat: add buildPartialDirBundle

### PR DESCRIPTION
**Pull request by connorlindsey**
### Description
Adds the `buildPartialDirBundle` and corresponding tests so that we can render the diff view for partials. 

Also enables the `render` tag in the liquid engine.

### Tasks
[KNO-6271](https://linear.app/knock/issue/KNO-6271/%5Bcli%5D-build-partial-bundle)

### CODE REVIEW

File: src/lib/helpers/liquid.ts [hunk 1]
```diff
@@ -17,7 +17,6 @@ const disabled = {
   },
 } as any;
 
-engine.registerTag("render", disabled);
```
Comment (connorlindsey) [inline src/lib/helpers/liquid.ts:20]: We'll need to enable this eventually since 'render' tags will be allowed. I think it's fine to just ship this now, but happy to hold off and do another PR/CLI release when the project is closer to shipping.

**Comment by thomaswhyyou** [comment]
Will look at this today!

**Review by thomaswhyyou — COMMENTED** [review]
LGTM! Mostly reviewed it in comparison to how we've implemented other modules, and this makes sense to me. Left a couple small comments/questions, but looks ready to go.

File: src/lib/helpers/liquid.ts (hunk 1 above)
Comment (connorlindsey) [reply src/lib/helpers/liquid.ts:20]: Sounds good, I'll remove that change. 

We support the exact same syntax as what liquidjs does. The only difference is that we expect a partial key instead of a file name, but they're still both just strings.

File: src/lib/marshal/partial/types.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export enum PartialType {
+  Html = "HTML",
+  Json = "JSON",
+  Markdown = "MARKDOWN",
+  Text = "TEXT",
+}
+
+// Partial payload data from the API.
+export type PartialData<A extends MaybeWithAnnotation = unknown> = A & {
```
[the remark this answers is not in the document]
Comment (connorlindsey) [reply src/lib/marshal/partial/types.ts:13]: Good call. I was thinking of holding off until we supported those fields more (e.g. `visual_block_enabled`) but might as well add them in now so we only have to update Control to support them later.

File: src/lib/helpers/liquid.ts (hunk 1 above)
Comment (thomaswhyyou) [reply src/lib/helpers/liquid.ts:20]: > The only difference is that we expect a partial key instead of a file name, but they're still both just strings.

It sounds some differences - like normally `.liquid` extension unless [`extname` option is set](https://liquidjs.com/api/interfaces/LiquidOptions.html#extname)? And you can use `with` or `for` tags in it?

**Review by thomaswhyyou — APPROVED** [review]
